zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/ElasticsearchRepository.java

```diff
@@ -187,6 +191,34 @@ public CompletableFuture<Void> reindexDocuments(
         .thenApplyAsync(ignored -> null, executor);
   }
 
+  private List<String> fetchMatchingIndexes(final String indexWildcard) {
+    final Pattern indexNamePattern = Pattern.compile(indexWildcard);
+    return client
+        .indices()
+        .get(new Builder().index(ALL_INDICES).build())
+        .join()
```

Comment:
:x: Do not join to avoid pinning the thread. See the comment in the OS reposiotry class.